### PR TITLE
feat: add PCM jobs endpoint

### PR DESCRIPTION
**Pull request by GeorgeBotzakis**
## Type

* ### Feature
  Added PCMJobs Endpoint (All, Get, GetJobErrors) under the PCM object in the SDK, alongside the appropriate type declaration file.
* ### Fix
  If a falsey version is provided in an endpoint, avoid adding an extra slash character before the next path parameter.
## Description

*A brief description of the goals of the pull request.*

## Issues
https://elasticpath.atlassian.net/browse/MT-12815

### CODE REVIEW

File: src/factories/request.js [hunk 1]
```diff
@@ -62,11 +62,14 @@ const fetchRetry = (
   new Promise((resolve, reject) => {
     const ver = version || config.version || ''
     config.auth.fetch
-      .bind()(`${config.protocol}://${config.host}/${ver}/${uri}`, {
-        method: method.toUpperCase(),
-        headers,
-        body: requestBody()
-      })
+      .bind()(
+        `${config.protocol}://${config.host}${ver ? `/${ver}` : ''}/${uri}`,
```
Comment (samblacklock) [inline src/factories/request.js:66]: 👏

File: src/factories/request.js [hunk 2]
```diff
@@ -62,11 +62,14 @@ const fetchRetry = (
   new Promise((resolve, reject) => {
     const ver = version || config.version || ''
```
Comment (field123) [inline src/factories/request.js:63]: Now that we check that `ver` is truthy do we need the ` || ''` here?

**Review by samblacklock — APPROVED** [review]
nice work @GeorgeBotzakis

File: src/factories/request.js (hunk 2 above)
Comment (GeorgeBotzakis) [reply src/factories/request.js]: Amended in new commit

**Pull request merged**